File: RayTracerVBSingleProj/StandardScenes.vb
Public Module StandardScenes

    Public ReadOnly DefaultScene As Scene = New Scene(DefaultThings(), DefaultLightSources(), DefaultCamera())

    Private Function DefaultThings() As Thing()
        Return New Thing() {
                    New Floor(StandardSurfaces.CheckerBoard),
                    New Sphere(New Vector3(0, 0.5, 0), 0.5, StandardSurfaces.Shiny),
                    New Sphere(New Vector3(1, 1, 1), 0.2, StandardSurfaces.Matt),
                    New Sphere(New Vector3(-2, 1, -1), 1, StandardSurfaces.Shiny)}
    End Function

    Private Function DefaultLightSources() As LightSource()
        Return New LightSource() {
                    New LightSource(New Vector3(-2, 2.5, 0), New Colour(0.49, 0.07, 0.07)),
                    New LightSource(New Vector3(1.5, 2.5, 1.5), New Colour(0.07, 0.07, 0.49)),
                    New LightSource(New Vector3(1.5, 2.5, -1.5), New Colour(0.07, 0.49, 0.071)),
                    New LightSource(New Vector3(0, 3.5, 0), New Colour(0.21, 0.21, 0.35))}
    End Function

    Private Function DefaultCamera() As Camera
        Return New Camera(New Vector3(0, 1, -6), New Vector3(0, 0, 0))
    End Function
End Module

File: RayTracerCSSingleProj/StandardScenes.cs
﻿namespace RayTracer
{
    public static class StandardScenes
    {
        public static readonly Scene DefaultScene = new Scene(DefaultThings(), DefaultLightSources(), DefaultCamera());

        private static Thing[] DefaultThings()
        {
            return new Thing[] {
                                new Floor(StandardSurfaces.CheckerBoard),
                                new Sphere(new Vector3(0,0.5,0),0.5,StandardSurfaces.Shiny),
                                new Sphere(new Vector3(1,1,1),0.2,StandardSurfaces.Matt),
                                new Sphere(new Vector3(-2,1,-1),1,StandardSurfaces.Shiny)
                                };
        }

        private static LightSource[] DefaultLightSources()
        {
            return new LightSource[] {
                                new LightSource(new Vector3(-2,2.5,0), new Colour(.49,.07,.07)),
                                new LightSource(new Vector3(1.5,2.5,1.5),new Colour(.07,.07,.49)),
                                new LightSource(new Vector3(1.5,2.5,-1.5),new Colour(.07,.49,.071)),
                                new LightSource(new Vector3(0,3.5,0),new Colour(.21,.21,.35))
                    };
        }

        private static Camera DefaultCamera()
        {
            return new Camera(new Vector3(0, 1, -6), new Vector3(0, 0, 0));
        }
    }
}
